src/Jobs/Notifications/AbstractItemNotificationJob.php

```diff
@@ -0,0 +1,122 @@
+<?php
+declare(strict_types=1);
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Jobs\Notifications;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Coupon\CouponHelper;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\ProductHelper;
+use Automattic\WooCommerce\GoogleListingsAndAds\Value\NotificationStatus;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class AbstractItemNotificationJob
+ * Generic class for the Notification Jobs containing items
+ *
+ * @since x.x.x
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Jobs\Notifications
+ */
+abstract class AbstractItemNotificationJob extends AbstractNotificationJob {
+
+	/**
+	 * Logic when processing the items
+	 *
+	 * @param array $args Arguments with the item id and the topic
+	 */
+	protected function process_items( array $args ): void {
+		if ( ! isset( $args['item_id'] ) || ! isset( $args['topic'] ) ) {
+			do_action(
+				'woocommerce_gla_error',
+				'Error sending the Notification. Topic and Item ID are mandatory',
+				__METHOD__
+			);
+			return;
+		}
+
+		$item  = $args['item_id'];
+		$topic = $args['topic'];
+
+		if ( $this->can_process( $item, $topic ) && $this->notifications_service->notify( $topic, $item ) ) {
+			$this->set_status( $item, $this->get_after_notification_status( $topic ) );
+			$this->maybe_mark_as_unsynced( $topic, $item );
+		}
+	}
+
+	/**
+	 * Set the notification status for the item.
+	 *
+	 * @param int    $item_id
+	 * @param string $status
+	 */
+	protected function set_status( int $item_id, string $status ): void {
+		$item = $this->get_item( $item_id );
+		$this->get_helper()->set_notification_status( $item, $status );
+	}
+
+	/**
+	 * Get the Notification Status after the notification happens
+	 *
+	 * @param string $topic
+	 * @return string
+	 */
+	protected function get_after_notification_status( string $topic ): string {
+		if ( str_contains( $topic, '.create' ) ) {
+			return NotificationStatus::NOTIFICATION_CREATED;
+		} elseif ( str_contains( $topic, '.delete' ) ) {
+			return NotificationStatus::NOTIFICATION_DELETED;
+		} else {
+			return NotificationStatus::NOTIFICATION_UPDATED;
+		}
+	}
+
+	/**
+	 * Checks if the item can be processed based on the topic.
+	 * This is needed because the item can change the Notification Status before
+	 * the Job process the item.
+	 *
+	 * @param int    $item_id
+	 * @param string $topic
+	 * @return bool
+	 */
+	protected function can_process( int $item_id, string $topic ): bool {
+		$item = $this->get_item( $item_id );
+
+		if ( str_contains( $topic, '.create' ) ) {
```

Comment:
To echo the comment https://github.com/woocommerce/google-listings-and-ads/pull/2278#discussion_r1504297176 by @martynmjones, I think it might be a bit safer to check if this is the allowed topics defined in NotificationService.php. Perhaps we could add an array `ALLOWED_TOPICS` in NotificaitonService.php?